dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/BaseAnalyticalObject.java

```diff
@@ -746,22 +746,30 @@ private void setPeriodNames( List<Period> periods, boolean dynamicNames, I18nFor
      */
     public void clear()
     {
-        dataDimensionItems.clear();
-        periods.clear();
+        clearIfNotNull( dataDimensionItems );
```

Comment:
Just wondering, what is the reason for checking for null? The idea is that want collections to never be null to avoid nullpointers throughout the code, and hence initialize the collections to empty ones initially in the beginning of this class.

Reply:
Fixed. Code reverted